### ISSUE TITLE
Two Factor Authentication

### ISSUE THREAD

**Issue by MuhammetDilmac** [activeadmin/activeadmin#4668]
Hello,
Anybody implemented two factor authentication with Google Authenticator on active admin login?

**Comment by timoschilling** [comment]
Please ask questions like this on Stackoverflow, the issue tracker is for bugs and so on.

Since ActiveAdmin Auth is based on devise, maybe one of this will help you:
- https://github.com/AsteriskLabs/devise_google_authenticator
- https://github.com/tinfoil/devise-two-factor
- https://github.com/zquestz/omniauth-google-oauth2

**Comment by bbozo** [comment]
It would be a nice extension to the docs, on how to integrate this, and which one to integrate for maximum compatibility and minimum pain, it's something only experience gives you

**Comment by timoschilling** [comment]
@bbozo sorry that is out of scope of AA, thats the domain of the devise gem

**Comment by varyonic** [comment]
We use active_model_otp with activeadmin.  Implementation was relatively straightforward, though the screen we added for the authcode after login was ugly.  I have nothing to publish or share anytime soon. We looked at the other gems Timo mentioned and they appeared more or less the same.  I agree StackOverflow is the place for this, though I can find nothing there.  I would love to be able to reference a blog post on the wiki if anyone writes one.
